src/main.rs

```diff
@@ -497,6 +553,183 @@ fn skip_token(buf: &[u8]) -> usize {
     }
 }
 
+#[derive(Default, PartialEq, Eq)]
+struct HunkHeader {
+    // range are (ofs,len) for the interval [ofs, ofs + len)
+    minus_range: (usize, usize),
+    plus_range: (usize, usize),
+}
+
+const WIDTH: [usize; 21] = [
+    0,
+    9,
+    99,
+    999,
+    9999,
+    99999,
+    999999,
+    9999999,
+    99999999,
+    999999999,
+    9999999999,
+    99999999999,
+    999999999999,
+    9999999999999,
+    99999999999999,
+    999999999999999,
+    9999999999999999,
+    99999999999999999,
+    999999999999999999,
+    9999999999999999999,
+    usize::max_value(),
+];
+
+fn width1(x: usize) -> usize {
+    let result = WIDTH[..WIDTH.len() - 1].binary_search(&x);
+    match result {
```

Comment:
I suggest you pull from https://github.com/mookid/diffr/commit/d220e99bd55060c48e12430a4dabdd7bb3e7bc61 ;)